### DISCUSSION TITLE
Is there a way to append a transform to a zod schema?

### DISCUSSION THREAD

**Discussion by Greg-Freeman** [kubb-labs/kubb#1229] [Q&A]
Currently we receive DTO's that we need to reshape and we would like to define zod transforms on the end of the schema to map between objects, and have the resulting type used in react query instead. Is there any way to do that, or would we need to use templates if we wanted it part of the automated generation?

Effectively i want something like this pseudo code to be output
getPetByIdQueryResponseSchema.transform(petDTO=> pet).parse() where pet is stored in the react query cache instead of petDTO

**Comment by stijnvanhulle** [comment]
In Kubb you can use 'parser' to use Zod in react-query: https://www.kubb.dev/plugins/swagger-tanstack-query/#parser. Could that help out or is more needed?

**Reply by Greg-Freeman** [reply]
> In Kubb you can use 'parser' to use Zod in react-query: https://www.kubb.dev/plugins/swagger-tanstack-query/#parser. Could that help out or is more needed?

I don't think that helps because my understanding is that will create a zod schema for me based on the swagger documentation, and then insert that into the generated react query code. That is not what i want though. I want a zod schema that maps data from the shape defined in the swagger documentation into _another object_ which will be the domain layer one used throughout my app. 

What i essentially want is the [select functionality](https://tanstack.com/query/v5/docs/framework/react/reference/useQuery) provided by react query

If i can provide my own zod schemas that just get inserted by the plugin into the react query generated code, that would work as well.

**Comment by Greg-Freeman** [comment]
@stijnvanhulle  Bumping this again. Is there anything I can do to map data from a DTO to an object I want stored in the query cache? This is happening because the API I'm using has responses that look like this:

```
{
  data: T,
  status: number,
  succeeded: boolean
 }
```
 
 I don't want to store the entire object, I just want to map data and store that in the query cache. But without a selector I have to write wrappers for _**every**_ hook that kubb generates
